lib/beaker-puppet/install_utils/puppet5.rb

```diff
@@ -0,0 +1,128 @@
+
+        logger.debug("YAML HASH BELOW:")
+        logger.debug(file_hash)
+        logger.debug("PLATFORM_DATA BELOW:")
+        logger.debug(file_hash[:platform_data])
+        return sha_yaml_folder_url, file_hash[:platform_data]
+      end
+
+      # get the build artifact URL for a host
+      #
+      # @param [Host] host Host to get artifact URL for
+      # @param [Hash] build_details Details of the build in a hash
+      # @param [String] build_url URL to the build
+      #
+      # @return [String] URL to the build artifact
+      def hosts_artifact_url(host, build_details, build_url)
+        packaging_platform = host[:packaging_platform]
+        if packaging_platform.nil?
+          message = <<-EOF
+            :packaging_platform not provided for host '#{host}', platform '#{host[:platform]}'
+            :packaging_platform should be the platform-specific key from this list:
+              #{ build_details.keys }
+          EOF
+          fail_test( message )
+        end
+
+        logger.debug("PLATFORM SPECIFIC INFO for #{host} (packaging name '#{packaging_platform}'):")
+        packaging_data = build_details[packaging_platform]
+        logger.debug("- #{ packaging_data }, isnil? #{ packaging_data.nil? }")
+        artifact_buildserver_path = packaging_data[:artifact]
+        fail_test('no artifact_buildserver_path found') if artifact_buildserver_path.nil?
+
+        artifact_url = "#{build_url}/#{artifact_buildserver_path}"
+        artifact_url_correct = link_exists?( artifact_url )
+        logger.debug("- artifact url: '#{artifact_url}'. Exists? #{artifact_url_correct}")
+        fail_test('artifact url built incorrectly') if !artifact_url_correct
+        artifact_url
+      end
+
+      # install build artifact on the given host
+      #
+      # @param [Host] host Host to install artifact on
+      # @param [String] project_name Name of project for artifact. Needed for OSX installs
+      # @param [String] artifact_url URL of the project install artifact
+      #
+      # @return nil
+      def install_artifact_on(host, artifact_url, project_name)
+        variant, _, _, _ = host[:platform].to_array
+        onhost_package_file = nil
+        if variant == 'solaris'
+          artifact_filename = File.basename( artifact_url )
+          artifact_folder = File.dirname( artifact_url )
+          fetch_http_file( artifact_folder, artifact_filename, '.' )
+          scp_to host, artifact_filename, '/tmp'
+          onhost_package_file = "/tmp/#{ artifact_filename }"
+        elsif variant == 'debian'
+          onhost_package_file = '/tmp/puppet.deb'
+          on host, "wget -O #{ onhost_package_file } #{ artifact_url }"
+        elsif variant == 'osx'
+          on host, "curl -O #{ artifact_url }"
+          onhost_package_file = "#{ project_name }*"
+        end
+
+        if onhost_package_file
+          host.install_local_package( onhost_package_file )
+        else
+          host.install_package( artifact_url )
```

Comment:
Trying this on ubuntu-16.04-amd64, this method ends up calling:

```
q2p1dby68kwqdaq.delivery.puppetlabs.net (q2p1dby68kwqdaq.delivery.puppetlabs.net) 14:06:34$ apt-get install --force-yes  -y http://builds.delivery.puppetlabs.net/puppet-agent/f0005a296b2e1f5d6a7541aac005d41550690bc5/artifacts/./deb/xenial/PC1/puppet-agent_5.0.0.137.gf0005a2-1xenial_amd64.deb
    Reading package lists...
    Building dependency tree...
    Reading state information...
    E: Unable to locate package http://builds.delivery.puppetlabs.net/puppet-agent/f0005a296b2e1f5d6a7541aac005d41550690bc5/artifacts/./deb/xenial/PC1
    E: Couldn't find any package by glob 'http://builds.delivery.puppetlabs.net/puppet-agent/f0005a296b2e1f5d6a7541aac005d41550690bc5/artifacts/./deb/xenial/PC1'
    E: Couldn't find any package by regex 'http://builds.delivery.puppetlabs.net/puppet-agent/f0005a296b2e1f5d6a7541aac005d41550690bc5/artifacts/./deb/xenial/PC1'
```

But it fails because `apt-get` can't seem to install the `.deb` from a URL. Should we curl/wget the `deb` locally to the SUT first?